Neos.Flow/Classes/Configuration/Loader/PolicyLoader.php

```diff
@@ -0,0 +1,99 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Configuration\Loader;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Configuration\Source\YamlSource;
+use Neos\Flow\Core\ApplicationContext;
+use Neos\Utility\Arrays;
+
+class PolicyLoader implements LoaderInterface
+{
+
+    /**
+     * @var YamlSource
+     */
+    private $yamlSource;
+
+    /**
+     * An absolute file path to store configuration caches in. If null no cache will be active.
+     *
+     * @var string
+     */
+    protected $temporaryDirectoryPath;
+
+    public function __construct(YamlSource $yamlSource)
+    {
+        $this->yamlSource = $yamlSource;
+    }
+
+    /**
+     * Set an absolute file path to store configuration caches in. If null no cache will be active.
+     *
+     * @param string $temporaryDirectoryPath
+     */
+    public function setTemporaryDirectoryPath(string $temporaryDirectoryPath): void
```

Comment:
Since the loader will likely fail without having set this path, we could maybe make it a constructor dependency. Would also make instanciation a bit easier at https://github.com/neos/flow-development-collection/pull/2449/files#diff-83398c2914c2ce46b6e55c9821468814652a275d332692492bb6edbbb84607f9R245-R247

Reply:
The only reason, the temporary directory path is needed in the policy loader is for behavior testing. The path is not used in development or production. This is kind of nasty workaround - which also exists in the current ConfigurationManager.

The temporary Policy.yaml is written [here](https://github.com/neos/flow-development-collection/blob/c00e0daf1d1df64fa073d67b25753a504f1b9a95/Neos.Flow/Tests/Behavior/Features/Bootstrap/SecurityOperationsTrait.php#L95). A few weeks ago I and @bwaidelich tried to fix this with something like a closure policy loader, but the ConfigurationManager just ignored them. This was related to how thows behaviour tests are creating an instance of the ConfigurationManager. Tbh, it was very complicated to understand why the test didn't work out and finally I couldn't fix this in a better way, than it is currently implemented.

So, this kinda nasty testing fix with a temporary policy file is still there and I would argue to keep the constructor for now, because I think someone with more insight in those behavior test might be able to fix this workaround - maybe with a custom yamlsource or such? And if the test doesn't need a temporary Policy.yaml we don't need to change the constructor.